Automattic/wp-calypso
# Stats: Date control - style new chart header with CSS

Related to #82082

## Proposed Changes

* This PR adds styling to the DatePicker dropdown button (font style, border, border radius), the Interval Dropdown button (font style, border, border radius) and the Interval Dropdown popover content (alignment, widths, padding, link style etc) to transform the standard Gutenberg dropdown component & the Interval component inside to match our designs. 

* This involves quite a lot of CSS overriding of the Intervals component styles. I've chosen this path for now as we do not want to interfere with the existing styling of the interval buttons. I suspect in the future, this may not be the best approach. But I'm open to suggestions! I think for now, it is. 

* Note: This PR does not address the final location of the two drop downs, nor the icons still needing to be added. These will come in a followup. 

* Also note: this isn't perfect yet. I hope it's a whole lot closer than it was, but I didn't want to sit on this PR any longer without some feedback. Further iterations will be needed to bring this exactly to match the designs. 

## Testing Instructions



* Open the Calypso live branch
* Navigate to `/stats/day/{site URL}`
* Confirm that everything remains unchanged without the feature flag applied
* Append `?flags=stats/date-control` to the end of the URL
* Confirm all of the updated styling matches the designs in c6yabERlJE06tKLECfpnch-fi-7781%3A4575

| Before  | After | Design |
| ------------- | ------------- | ------------- |
| ![image](https://github.com/Automattic/wp-calypso/assets/30754158/a17d7dc1-59a2-4c27-984c-75cd8b74f44a)  | ![image](https://github.com/Automattic/wp-calypso/assets/30754158/d7c13389-3f62-4889-a680-a96c690942e4)  | ![image](https://github.com/Automattic/wp-calypso/assets/30754158/80cfabd4-3363-4426-bd1b-38814522a068)  |

## Pre-merge Checklist
